File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    const confirmArray = [];
    const bracketsArray = ["(", "{", "[", ")", "}", "]"]

    for (let i = 0; i < s.length; i++) {
      if (s[i] === bracketsArray[0] || s[i] === bracketsArray[1] || s[i] === bracketsArray[2]) {
        confirmArray.push(s[i]);
      } else {
        if (bracketsArray.indexOf(s[i]) - bracketsArray.indexOf(confirmArray[confirmArray.length - 1]) === 3) {
          confirmArray.pop();
        } else {
          return false;
        }
      }
    }
  
    return confirmArray.length === 0;
};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        BRAKET_DICT = {
            "]": "[",
            ")": "(",
            "}": "{"
        }
        
        stack_list = []
        for braket in s:
            if braket in BRAKET_DICT:
                if len(stack_list) == 0  or stack_list.pop() != BRAKET_DICT[braket]:
                    return False
            else:
                stack_list.append(braket)
        
        if len(stack_list) != 0:
            return False
        else:
            return True